### PR TITLE
gh-212: metadata for bias computation

### PR DESCRIPTION
**Pull request by JaimeRZP**
Adds more metadata to the maps' fields used to compute the Jacknife covariance matrix. 
_The history of the branch seems linked to my previous PR??_

### CODE REVIEW

**Comment by ntessore** [comment]
Try `git checkout more_metadata && git rebase --onto main FootprintFilter more_metadata && git push -f`?

**Comment by JaimeRZP** [comment]
> Try `git checkout more_metadata && git rebase --onto main FootprintFilter more_metadata && git push -f`?

Nice git arcane magic!

File: heracles/fields.py [hunk 1]
```diff
@@ -305,10 +305,14 @@ async def __call__(
             del vis
 
         # compute bias of positions, including weight variance
-        bias = ngal / (4 * np.pi) * mapper.area**2 * (w2mean / nbar**2)
+        variance = w2mean / wmean**2
+        neff = (npix * nbar) / (4 * np.pi * wmean)
```
Comment (ntessore) [inline heracles/fields.py:309]: I think for positions this needs to be
```suggestion
        variance = 1.0
        neff = (nbar / mapper.area)**2 / (ngal / (4 * np.pi * fsky)) / w2mean
```

File: heracles/fields.py [hunk 2]
```diff
@@ -371,10 +375,14 @@ async def __call__(
         val /= wbar
 
         # compute bias from variance (per object)
-        bias = 4 * np.pi * fsky**2 * (var / wmean**2) / ngal
+        variance = var / wmean**2
+        neff = ngal / (4 * np.pi * fsky)
+        bias = fsky * variance / neff
```
Comment (ntessore) [inline heracles/fields.py:380]:
```suggestion
        variance = var / w2mean
        deff = w2mean / wmean**2
        neff = ngal / (4 * np.pi * fsky) / deff
        bias = fsky * variance / neff
```

File: heracles/fields.py [hunk 3]
```diff
@@ -441,10 +449,14 @@ async def __call__(
         val /= wbar
 
         # bias from measured variance, for E/B decomposition
-        bias = 2 * np.pi * fsky**2 * (var / wmean**2) / ngal
+        variance = var / wmean**2
+        neff = ngal / (2 * np.pi * fsky)  # should we include the factor of 2 here?
+        bias = fsky * variance / neff
```
Comment (ntessore) [inline heracles/fields.py:454]:
```suggestion
        variance = var / w2mean
        deff = w2mean / wmean**2
        neff = ngal / (2 * np.pi * fsky) / deff
        bias = fsky * variance / neff / 2
```

**Review by ntessore — CHANGES_REQUESTED** [review]
The `Weights` class doesn't have the extra metadata yet. We should also

1. figure out better names for the metadata, particularly `variance`, which isn't the variance;
2. figure out whether to store density and effective density separately.

Let's discuss offline.

File: heracles/fields.py (hunk 1 above)
Comment (JaimeRZP) [reply heracles/fields.py]: I don't think I understand these last two suggestions.
In the Scalar and complex fields we don't define w2mean. 
Instead we have wmean and var.

**Review by ntessore — APPROVED** [review]
Excellent work 🎉

**Pull request merged**